### DISCUSSION TITLE
Enhancement: Canvas-like feature

### DISCUSSION THREAD

**Discussion by djuillard** [danny-avila/LibreChat#4600] [Feature Requests & Suggestions]
### What features would you like to see added?

I would like to have the Canvas feature like to one proposed in ChatGPT Plus : https://openai.com/index/introducing-canvas/

### More details

This feature enables to interact with code (or text) while highlighting some code snippets (or text extracts) in a side panel (similar to the kind of side panel appearing when using the artifacts). This enables to highlight and then query what we just highlighted so that we can ask for edition, and the AI only applies the modification to the part we wanted to focus on. It does not re-generate the full text or code.

### Which components are impacted by your request?

UI

### Pictures

https://openai.com/index/introducing-canvas/

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by rsexton404** [comment]
ICYMI: The LangChain folks have created a very nice open-source implementation that they call [Open Canvas](https://github.com/langchain-ai/open-canvas). I'd love to see a lightweight Markdown editor like this integrated into LibreChat, which would unlock a lot of content creation use cases for writers.

**Reply by djuillard** [reply]
Good catch, thanks.

**Reply by vanguardbi** [reply]
Any updates here?

**Comment by Defiect** [comment]
Is this on the roadmap for 2026?
